agent/consul/state/catalog.go

```diff
@@ -3336,7 +3344,9 @@ func getTermGatewayVirtualIPs(tx WriteTxn, services []structs.LinkedService, ent
 	addrs := make(map[string]structs.ServiceAddress, len(services))
 	for _, s := range services {
 		sn := structs.ServiceName{Name: s.Name, EnterpriseMeta: *entMeta}
-		vip, err := assignServiceVirtualIP(tx, sn)
+		// TODO(peering): should this take peername?
```

Comment:
No it shouldn't take a peer name. Can you pass in structs.DefaultPeerKeyword here?

Terminating gateway virtual IPs are for transparent proxies to dial services represented by a terminating gateway. But terminating gateways can't route to services in peered clusters.